packages/streamr-logger/include/streamr-logger/StreamrLogFormatter.hpp

```diff
@@ -0,0 +1,175 @@
+#ifndef STREAMER_LOGGER_STREAMRLOGFORMATTER_H
+#define STREAMER_LOGGER_STREAMRLOGFORMATTER_H
+
+#include <iostream>
+#include <folly/Format.h>
+#include <folly/logging/LogFormatter.h>
+#include <folly/logging/LogLevel.h>
+
+namespace {
+
+static constexpr int maxFileNameAndLineNumberLength{36};
+static constexpr folly::StringPiece fileNameAndLineNumberSeparator{": "};
+static constexpr auto separatorLength{
+    std::ssize(fileNameAndLineNumberSeparator)};
+static const std::string nonTruncatedFormatter =
+    "{}{}{} [{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}.{}] ({:<" +
+    std::to_string(maxFileNameAndLineNumberLength) + "}): {}{}{}\n";
+static constexpr std::string_view truncatedFormatter =
+    "{}{}{} [{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}.{}] ({: <*}{}{}): {}{}{}\n";
+
+struct LogLevelData {
+    folly::StringPiece logLevelName;
+    folly::StringPiece color;
+};
+
+// FATAL cannot be used in folly because it aborts. So CRITICAL is converted
+// to Streamr fatal
+constexpr LogLevelData getLogLevelData(const folly::LogLevel level) {
```

Comment:
name function more clearly to communicate intent, what data?

Reply:
add a comment about why switch statement is needed (std::map cannot be used in compile time)